src/Adapter/Smarty/SmartyCacheConfiguration.php

```diff
@@ -93,4 +104,50 @@ public function validateConfiguration(array $configuration)
             $configuration['smarty_console_key']
         );
     }
+
+    private function getSmartyCachingType(): string
+    {
+        $content = false;
```

Comment:
I don't see why you read the whole file to get the config, maybe I am missing something. If not, then it may be exists a 'simpler' solution like `return defined(_PS_SMARTY_CACHING_TYPE_) ? _PS_SMARTY_CACHING_TYPE_ : 'filesystem'; `

Reply:
I guess you're right, I just repeated what have been done for `_PS_MODE_DEV_` but yeah, definitely seems a better way to check the const directly!

Reply:
@atomiix could you improve it? Let's make sure to set it to `filesystem` even if someone modifies the file and sets `whatever` :-)

Reply:
@kpodemski we still allow mysql here though